src/starknet.rs

```diff
@@ -1630,3 +1627,37 @@ pub(crate) mod handler {
         }
     }
 }
+
+thread_local!(pub static SYSCALL_HANDLER_VTABLE: std::cell::Cell<*mut ()>  = const { std::cell::Cell::new(null_mut()) });
+
+#[allow(non_snake_case)]
+/// Runtime function that calls the `cheatcode` syscall
+///
+/// The Cairo compiler doesn't specify that the cheatcode syscall needs the syscall handler,
+/// so a pointer to `StarknetSyscallHandlerCallbacks` is stored as a `thread::LocalKey` and accesed in runtime by this function.
+pub extern "C" fn cairo_native__vtable_cheatcode(
+    result_ptr: &mut ArrayAbi<Felt252Abi>,
+    selector: &Felt252Abi,
+    input: &ArrayAbi<Felt252Abi>,
+) {
+    let ptr = SYSCALL_HANDLER_VTABLE.with(|ptr| ptr.get());
+    if ptr.is_null() {
+        *result_ptr = ArrayAbi {
+            ptr: null_mut(),
+            since: 0,
+            until: 0,
+            capacity: 0,
+        };
+        return;
+    }
```

Comment:
Why not an assert? The way I see it, if the cheatcode is called and there's not a valid syscall handler there's not really much to do other than crash.

Reply:
You're right. I'm going to change it to an assert.

Reply:
Done